testnet-contracts/test/test_cosmosBridge.js

```diff
@@ -0,0 +1,356 @@
+
+require("chai")
+  .use(require("chai-as-promised"))
+  .use(require("chai-bignumber")(BigNumber))
+  .should();
+
+contract("CosmosBridge", function(accounts) {
+  // System operator
+  const operator = accounts[0];
+
+  // Initial validator accounts
+  const userOne = accounts[1];
+  const userTwo = accounts[2];
+  const userThree = accounts[3];
+
+  // Contract's enum ClaimType can be represented a sequence of integers
+  const CLAIM_TYPE_BURN = 0;
+  const CLAIM_TYPE_LOCK = 1;
+
+  describe("CosmosBridge smart contract deployment", function() {
+    beforeEach(async function() {
+      // Deploy Valset contract
+      this.initialValidators = [userOne, userTwo, userThree];
+      this.initialPowers = [5, 8, 12];
+      this.valset = await Valset.new(
+        operator,
+        this.initialValidators,
+        this.initialPowers
+      );
+
+      // Deploy CosmosBridge contract
+      this.cosmosBridge = await CosmosBridge.new(operator, this.valset.address);
+
+      // Deploy Oracle contract
+      this.oracle = await Oracle.new(
+        operator,
+        this.valset.address,
+        this.cosmosBridge.address
+      );
+
+      // Deploy BridgeBank contract
+      this.bridgeBank = await BridgeBank.new(
+        operator,
+        this.oracle.address,
+        this.cosmosBridge.address
+      );
+    });
+
+    it("should deploy the CosmosBridge with the correct parameters", async function() {
+      this.cosmosBridge.should.exist;
+
+      const claimCount = await this.cosmosBridge.prophecyClaimCount();
+      Number(claimCount).should.be.bignumber.equal(0);
+
+      const cosmosBridgeValset = await this.cosmosBridge.valset();
+      cosmosBridgeValset.should.be.equal(this.valset.address);
+    });
+
+    it("should allow the operator to set the Oracle", async function() {
+      this.oracle.should.exist;
+
+      await this.cosmosBridge.setOracle(this.oracle.address, {
+        from: operator
+      }).should.be.fulfilled;
+
+      const bridgeOracle = await this.cosmosBridge.oracle();
+      bridgeOracle.should.be.equal(this.oracle.address);
+    });
+
+    it("should not allow the operater to update the Oracle once it has been set", async function() {
+      await this.cosmosBridge.setOracle(this.oracle.address, {
+        from: operator
+      }).should.be.fulfilled;
+
+      await this.cosmosBridge
+        .setOracle(this.oracle.address, {
+          from: operator
+        })
+        .should.be.rejectedWith(EVMRevert);
+    });
+
+    it("should allow the operator to set the Bridge Bank", async function() {
+      this.bridgeBank.should.exist;
+
+      await this.cosmosBridge.setBridgeBank(this.bridgeBank.address, {
+        from: operator
+      }).should.be.fulfilled;
+
+      const bridgeBank = await this.cosmosBridge.bridgeBank();
+      bridgeBank.should.be.equal(this.bridgeBank.address);
+    });
+
+    it("should not allow the operater to update the Bridge Bank once it has been set", async function() {
```

Comment:
```suggestion
    it("should not allow the operator to update the Bridge Bank once it has been set", async function() {
```

Reply:
Fixed in https://github.com/cosmos/peggy/pull/82/commits/ed748c7f8ed2bd377feb8be11969e76fde0f0773